mcpgateway/plugins/framework/models.py

```diff
@@ -662,17 +668,32 @@ class GlobalContext(BaseModel):
     user: Optional[str] = None
     tenant_id: Optional[str] = None
     server_id: Optional[str] = None
+    state: dict[str, Any] = {}
+    metadata: dict[str, Any] = {}
 
 
-class PluginContext(GlobalContext):
+class PluginContext(BaseModel):
     """The plugin's context, which lasts a request lifecycle.
 
     Attributes:
-       metadata: context metadata.
        state:  the inmemory state of the request.
+       global_context: the context that is shared across plugins.
+       metadata: plugin meta data.
+
+    Examples:
+        >>> gctx = GlobalContext(request_id="req-123")
+        >>> ctx = PluginContext(global_context=gctx)
+        >>> ctx.global_context.request_id
+        'req-123'
+        >>> ctx.global_context.user is None
+        True
+        >>> ctx.state["somekey"] = "some value"
+        >>> ctx.state["somekey"]
+        'some value'
     """
 
     state: dict[str, Any] = {}
```

Comment:
Similar issue here (state, metadata) with mutable dict assignments. Change to default_factory.

Reply:
done!